src/debugger/ios/iOSPlatform.ts

```diff
@@ -35,28 +35,17 @@ export class IOSPlatform implements IAppPlatform {
                 runArguments.push(this.simulatorTarget);
             }
 
-            const runIos = new CommandExecutor(this.projectPath).spawnReactCommand("run-ios", runArguments);
+            const runIosSpawn = new CommandExecutor(this.projectPath).spawnChildReactCommandProcess("run-ios", runArguments);
             const deferred = Q.defer<void>();
-            runIos.on("error", (err: Error) => {
-                deferred.reject(err);
-            });
-            runIos.stderr.on("data", (data: Buffer) => {
+            runIosSpawn.stderr.on("data", (data: Buffer) => {
                 const dataString = data.toString();
                 if (dataString.indexOf("No devices are booted") !== -1 // No emulators are started
                     || dataString.indexOf("FBSOpenApplicationErrorDomain") !== -1) { // The incorrect emulator is started
                     deferred.reject(new Error("Unable to launch iOS simulator. Try specifying a different target."));
                 }
             });
-            runIos.on("exit", (code: number) => {
-                if (code !== 0) {
-                    const err = new Error(`Command failed with exit code ${code}`);
-                    Log.commandFailed(["react-native", "run-ios"].concat(runArguments).join(" "), err);
-                    deferred.reject(err);
-                } else {
-                    deferred.resolve(void 0);
-                }
-            });
-            return deferred.promise;
+
+            return Q.all<void>([deferred.promise, runIosSpawn.outcome.then(() => deferred.resolve(void 0))]).then(() => {});

```

Comment:
This line confuses me. You wait for the deferred to resolve, but you also resolve it when the second promise completes as well?

Really this method does not want or need a promise from the `CommandExecutor`, it is managing the child process lifecycle completely and dealing with the other promise just confuses things.